eth/stagedsync/stage_difficulty.go

```diff
@@ -98,12 +99,14 @@ func SpawnDifficultyStage(s *StageState, tx kv.RwTx, cfg DifficultyCfg, ctx cont
 				return err
 			}
 
-			td.Add(td, header.Difficulty)
-
-			if header.Eip3675 {
+			if header.Difficulty.Cmp(serenity.SerenityDifficulty) == 0 {
```

Comment:
SerenityDifficulty is 0, this works just fine when dealing with proof-of-work since difficulty is never 0, and on clique difficulty can be either 1 or 2, however i am not quite sure if there is any edge case where block difficulty is 0.

Reply:
On the PoW mainnet it can't be 0 because block difficulty is at least 0x20000 per the Yellow Paper.

Reply:
fair enough